sr/comp/cli/add_delay.py

```diff
@@ -25,10 +25,45 @@ def parse_duration(time_str):
     return timedelta(**time_params)
 
 def parse_datetime(when_str):
-    import timelib
+    import re
+    from datetime import datetime
+    from dateutil.parser import parse as parse_date
     from dateutil.tz import tzlocal
 
-    when = timelib.strtodatetime(when_str)
+    def parse_now(match):
+        return datetime.now()
+
+    def parse_future(match):
+        offset = parse_duration(match.group(1))
+        return datetime.now() + offset
+
+    def parse_past(match):
+        offset = parse_duration(match.group(1))
+        return datetime.now() - offset
+
+    def parse_absolute(match):
+        return parse_date(match.group(0))
+
+    DATETIME_PATTERNS = [
+        (r'^([^ ]+)\s*ago$', parse_past),
```

Comment:
This looks like it'll match `5mago` is that desired?

Reply:
That string's kind of janky but it is also unambiguous, I thought it was probably fine to let it through—thoughts?

Reply:
Yeah, let's :shipit:. If it turns out to be an issue we can make it more strict.